query-engine/connectors/mongodb-query-connector/src/query_builder/read_query_builder.rs

```diff
@@ -204,57 +172,21 @@ impl MongoReadQueryBuilder {
     }
 
     /// Finalizes the builder and builds a `MongoQuery`.
-    pub(crate) fn build(mut self) -> crate::Result<MongoReadQuery> {
+    pub(crate) fn build(mut self) -> crate::Result<ReadQuery> {
         self.finalize()?;
-
-        // Depending on the builder contents, either an
-        // aggregation pipeline or a plain query is build.
-        // if self.joins.is_empty()
-        //     && self.order_joins.is_empty()
-        //     && self.aggregations.is_empty()
-        //     && self.cursor_data.is_none()
-        // {
-        //     Ok(self.build_find_query())
-        // } else {
-        // }
-
         Ok(self.build_pipeline_query())
     }
 
-    /// Note: Mongo Driver broke usage of the simple API, can't be used by us anymore. Always doing aggr. pipeline for now.
-    /// Simplest form of find-documents query: `coll.find(filter, opts)`.
-    #[allow(dead_code)]
-    fn build_find_query(self) -> MongoReadQuery {
```

Comment:
I removed all code that a) was called transitively by this method, and b) called into this method but was never used.